src/Domain/Services/ExtendRestApi.php

```diff
@@ -18,13 +19,34 @@ class ExtendRestApi {
 	 */
 	private $package;
 
+	/**
+	 * Holds the formatters class instance.
+	 *
+	 * @var Formatters
+	 */
+	private $formatters;
+
 	/**
 	 * Constructor
 	 *
-	 * @param Package $package An instance of the package class.
+	 * @param Package    $package An instance of the package class.
+	 * @param Formatters $formatters An instance of the formatters class.
+	 */
+	public function __construct( Package $package, Formatters $formatters ) {
+		$this->package    = $package;
+		$this->formatters = $formatters;
+	}
+
+	/**
+	 * Magic getter for protected properties.
+	 *
+	 * @param string $name Property name.
 	 */
-	public function __construct( Package $package ) {
-		$this->package = $package;
+	public function __get( $name ) {
+		if ( in_array( $name, [ 'formatters' ], true ) ) {
+			return $this->$name;
+		}
+		return null;
 	}
```

Comment:
Using the getter method is clever, but I would prefer that there be an explicit getter for formatters. One problem with magic method usage like this is that IDEs can't resolve the methods so it makes the method more opaque. We can also explicitly add typing for what is returned which helps for things like automatic documentation (which Woo Core is using) and also IDE intellisense (i.e. `@returns FormatterInterface`)

Reply:
Replaced with `get_formatter( $name )`